crates/ra_ide/src/lib.rs

```diff
@@ -472,15 +472,39 @@ impl Analysis {
 
     /// Computes assists (aka code actions aka intentions) for the given
     /// position.
-    pub fn assists(&self, config: &AssistConfig, frange: FileRange) -> Cancelable<Vec<Assist>> {
+    pub fn unresolved_assists(
+        &self,
+        config: &AssistConfig,
+        frange: FileRange,
+    ) -> Cancelable<Vec<Assist>> {
+        self.with_db(|db| {
+            ra_assists::Assist::unresolved(db, config, frange)
+                .into_iter()
+                .map(|assist| Assist {
+                    id: assist.id,
+                    label: assist.label,
+                    group_label: assist.group.map(|it| it.0),
+                    source_change: None,
+                })
+                .collect()
+        })
+    }
+
+    /// Computes assists (aka code actions aka intentions) for the given
+    /// position.
+    pub fn resolved_assists(
+        &self,
+        config: &AssistConfig,
+        frange: FileRange,
+    ) -> Cancelable<Vec<Assist>> {
         self.with_db(|db| {
             ra_assists::Assist::resolved(db, config, frange)
                 .into_iter()
                 .map(|assist| Assist {
                     id: assist.assist.id,
                     label: assist.assist.label,
                     group_label: assist.assist.group.map(|it| it.0),
-                    source_change: assist.source_change,
+                    source_change: Some(assist.source_change),
                 })
                 .collect()
         })
```

Comment:
The signatures of these two methods should be `-> Cancelable<Vec<Assist>>` and `-> Cancelable<Vec<ResolvedAssist>>`